tests/integration/sdk_api/alerts/test_alerts.py

```diff
@@ -595,6 +595,46 @@ def test_alert_templates(self):
             alert_criteria=alert_criteria,
         )
 
+    def test_alert_activation_notification_state(self):
+        project_name = "my-new-project"
+        project = mlrun.new_project(project_name)
+
+        alert_name = "alert1"
+        alert_entity = alert_objects.EventEntityKind.JOB
+        event_name = alert_objects.EventKind.FAILED
+
+        failure_webhook_notification = mlrun.common.schemas.Notification(
+            name="webhook1",
+            kind="webhook",
+            params={"url": "incorrect-url"},
+            when=["error"],
+            message="fail",
+            condition="",
+        )
+        self._create_alert(
+            project_name,
+            alert_name,
+            alert_entity,
+            project_name,
+            "Alert summary",
+            event_name,
+            reset_policy=mlrun.common.schemas.alert.ResetPolicy.AUTO,
+            notifications=[
+                mlrun.common.schemas.AlertNotification(
+                    notification=failure_webhook_notification
+                )
+            ],
+        )
+        for i in range(1, 4):
```

Comment:
why 4? magic?

Reply:
@liranbg technically it is three, just to ensure that bug isn't reproduces and notification is updated very fast

Reply:
how can you ensure _notification is updated very fast_ ?
perhaps on slow ci system the test would transiently fail? not sure I understand the scenario maybe could you elaborate on test docstring?

Reply:
@liranbg we request alert_activation right after we posted an event on which this alert is created. And I expect to have notification state quite fast (it's not requirement, though a good thing) in activation, so in most cases it won't be really visible for user that notification is updated on the fly. Notification is send in async function which we not await for. 
> perhaps on slow ci system the test would transiently fail
 it may, then we might need to add sleep there
 
 I'll add a comment